src/daft-local-execution/src/sources/in_memory.rs

```diff
@@ -28,8 +30,11 @@ impl Source for InMemorySource {
         _runtime_handle: &mut ExecutionRuntimeHandle,
         _io_stats: IOStatsRef,
     ) -> crate::Result<SourceStream<'static>> {
-        let data = self.data.clone();
-        Ok(Box::pin(futures::stream::iter(data)))
+        if self.data.is_empty() {
+            let empty = Arc::new(MicroPartition::empty(Some(self.schema.clone())));
+            return Ok(Box::pin(futures::stream::iter(vec![empty])));
```

Comment:
we can avoid allocating a vec here. 

```suggestion
            return Ok(Box::pin(futures::stream::once(async { empty })));
```

Reply:
Thank you!